xclim/core/calendar.py

```diff
@@ -1588,3 +1589,302 @@ def get_doys(start, end, inclusive):
+
+    try:
+        starts = da[dim]
+        window = starts.attrs["window"]
+        stride = starts.attrs["stride"]
+        freq = starts.attrs["freq"]
+        unequal_lengths = bool(starts.attrs["unequal_lengths"])
+    except (AttributeError, KeyError) as err:
+        raise ValueError(
+            f"`unstack_periods` can't find the window, stride and freq attributes on the {dim} coordiantes."
+        ) from err
+
+    if unequal_lengths:
+        try:
+            lengths = da[f"{dim}_length"]
+        except KeyError as err:
+            raise ValueError(
+                f"`unstack_periods` can't find the `{dim}_length` coordinate."
+            ) from err
+        # Get length as number of points
+        m, u = infer_sampling_units(da.time)
+        lengths = lengths // m
+    else:
+        # It is acceptable to lose "{dim}_length" if they were all equal
+        lengths = xr.DataArray([da.time.size] * da[dim].size, dims=(dim,))
+
+    # Convert from the fake axis to the real one
+    time_as_delta = da.time - da.time[0]
+    if da.time.dtype == "O":
+        # cftime can't add with np.timedelta64 (restriction comes from numpy which refuses to add O with m8)
+        time_as_delta = pd.TimedeltaIndex(
+            time_as_delta
+        ).to_pytimedelta()  # this array is O, numpy complies
+    else:
+        # Xarray will return int when iterating over datetime values, this returns timestamps
+        starts = pd.DatetimeIndex(starts)
+
+    def _reconstruct_time(start):
+        times = time_as_delta + start
+        return xr.DataArray(times, dims=("time",), coords={"time": times}, name="time")
+
+    # Easy case:
+    if window == stride:
+        # just concat them all
+        periods = []
+        for i, (start, length) in enumerate(zip(starts.values, lengths.values)):
+            real_time = _reconstruct_time(start)
+            periods.append(
+                da.isel(**{dim: i}, drop=True)
+                .isel(time=slice(0, length))
+                .assign_coords(time=real_time.isel(time=slice(0, length)))
+            )
+        return xr.concat(periods, "time")
+
+    # Difficult and ambiguous case
+    if (window / stride) % 2 != 1:
+        raise NotImplementedError(
+            "`unstack_periods` can't work with strides that do not divide the window into an odd number of parts."
+            f"Got {window} / {stride} which is not an odd integer."
+        )
+
+    # Non-ambiguous overlapping case
+    Nwin = window // stride
+    mid = (Nwin - 1) // 2  # index of the center window
+
+    mult, *args = parse_offset(freq)
+    strd_frq = construct_offset(mult * stride, *args)
+
+    periods = []
+    for i, (start, length) in enumerate(zip(starts.values, lengths.values)):
+        real_time = _reconstruct_time(start)
+        slices = real_time.resample(time=strd_frq).groupers[0].group_indices
+        if i == 0:
+            slc = slice(slices[0].start, min(slices[mid].stop, length))
```

Comment:
A bit confused here. There should be a special case like this for all stride windows where the full window is not included in the full period of interest. Hope this example shows well what I mean: 

window=5
stride=1
==> mid = 2


For i=2,3,4, using mid=2 correctly selects the middle of the window
But for i=1, the "o" should be in position "1", not mid=2 ? 
Agaub for i =0, mid=2 is not appropriate, the middle or the window (with NaNs on left) is 0.
  | i   | relative position of "o" among "x"'s | Not in time | 0   | 1   | 2   | 3   | 4   | 5   | 6   |
  | --- | -------------------------------------------- | ----------: | --- | --- | --- | --- | --- | --- | --- |
  | 4   | 2                                            |             |     |     | x   | x   | o   | x   | x   |
  | 3   | 2                                            |             |     | x   | x   | o   | x   | x   |     |
  | 2   | 2                                            |             | x   | x   | o   | x   | x   |     |     |
  | 1   | 1                           |           Nan | x   | o   | x   | x   |     |     |     |
  | 0   | 0                            |        Nan  Nan | o   | x   | x   |     |     |     |     |

Reply:
So, more briefly, could it be that we need the condition: `i<mid` ?

Reply:
Hmm, I think your example is not possible with the current function. The first window (`i = 0`) is always complete. Maybe you expected something looking `rolling(..., center=True)`, that's not what I implemented.

If we don't touch the `min_length` arg and only keep full windows, we get:

  | i   | extracted indexes |  0   | 1   | 2   | 3   | 4   | 5   | 6   |
  | --- | --------------------------------------------  | --- | --- | --- | --- | --- | --- | --- |
  | 2   | 2:                                            |     |     | x   | x   | o   | o   | o   |
  | 1   | 2                                            |     | x   | x   | o   | x   | x   |     |
  | 0   | :2                                            | o   | o   | o   | x   | x   |     |     |

If we allow `min_length=4`:

  | i   | extracted indexes |  0   | 1   | 2   | 3   | 4   | 5   | 6   |
  | --- | --------------------------------------------  | --- | --- | --- | --- | --- | --- | --- |
  | 3   | 2:                                            |     |     |      | x   | x   | o   | o   |
  | 2   | 2                                            |     |     | x   | x   | o   | x   | x   |
  | 1   | 2                                             |     | x   | x   | o   | x   | x   |     |
  | 0   | :2                                            | o   | o   | o   | x   | x   |     |     |

It was easier to code it like this than to imitate `center=True`, but indeed that might be confusing...

Reply:
I think I'll add a table like this to the docstring!

Reply:
ah right, `rolling` by default does not center, oups.